src/data/set/finite.lean

```diff
@@ -584,32 +584,32 @@ lemma finite.card_to_finset {s : set α} [fintype s] (h : s.finite) :
 by { rw [← finset.card_attach, finset.attach_eq_univ, ← fintype.card], congr' 2, funext,
      rw set.finite.mem_to_finset }
 
-section
-
-local attribute [instance, priority 1] classical.prop_decidable
+section decidable_eq
 
-lemma to_finset_compl {α : Type*} [fintype α] (s : set α) :
+lemma to_finset_compl {α : Type*} [fintype α] [decidable_eq α] (s : set α) [decidable_pred (∈ s)] :
   sᶜ.to_finset = (s.to_finset)ᶜ :=
 by ext; simp
 
-lemma to_finset_inter {α : Type*} [fintype α] (s t : set α) :
+lemma to_finset_inter {α : Type*} [fintype α] [decidable_eq α] (s t : set α)
+  [decidable_pred s] [decidable_pred t] :
   (s ∩ t).to_finset = s.to_finset ∩ t.to_finset :=
 by ext; simp
 
-lemma to_finset_union {α : Type*} [fintype α] (s t : set α) :
+lemma to_finset_union {α : Type*} [fintype α] [decidable_eq α] (s t : set α)
+  [decidable_pred s] [decidable_pred t] :
   (s ∪ t).to_finset = s.to_finset ∪ t.to_finset :=
 by ext; simp
 
-lemma to_finset_ne_eq_erase {α : Type*} [fintype α] (a : α) :
+lemma to_finset_ne_eq_erase {α : Type*} [fintype α] [decidable_eq α] (a : α) :
   {x : α | x ≠ a}.to_finset = finset.univ.erase a :=
 by ext; simp
 
-lemma card_ne_eq [fintype α] (a : α) :
+lemma card_ne_eq [fintype α] [decidable_eq α] (a : α) :
```

Comment:
I think this one should maybe have `[fintype {x : α | x ≠ a}]` too, since that's the one needed to state the LHS of the lemma.

Reply:
OK!